app/services/base_service.rb

```diff
@@ -155,7 +155,7 @@ def self.call_async(*, **, &block)
   def initialize(current_user = nil)
     @result = Result.new
     @source = CurrentContext&.source
-    result.user = current_user
+    result.user = current_user.is_a?(User) ? current_user : nil
```

Comment:
Interesting, it is some legacy behavior. Not sure we are still using this user.
Maybe a good refact: Only deal with the current_user when we need it. Otherwise, looks good ;)

Reply:
100% agree 👍